tidy3d/components/medium.py

```diff
@@ -71,6 +74,75 @@ def _eps_model(self, frequency: float) -> complex:
 """ Medium Definitions """
 
 
+class NonlinearSpec(ABC, Tidy3dBaseModel):
+    """Abstract specification for adding a nonlinearity to a medium.
+
+     Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for strong nonlinearities. Increasing `numiters` can help with convergence.
+    """
+
+    numiters: pd.PositiveInt = pd.Field(
+        1,
+        title="Number of iterations",
+        description="Number of iterations for solving nonlinear constitutive relation.",
+    )
+
+    @pd.validator("numiters", always=True)
+    def _validate_numiters(cls, val):
+        """Check that numiters is not too large."""
+        if val > NONLINEAR_MAX_NUMITERS:
+            raise ValidationError(
+                "'NonlinearSpec.numiters' must be less than "
+                f"{NONLINEAR_MAX_NUMITERS}, currently {val}."
+            )
+        return val
+
+
+class NonlinearSusceptibility(NonlinearSpec):
+    """Specification adding an instantaneous nonlinear susceptibility to a medium.
+
+    Note
+    ----
+    The instantaneous nonlinear polarization is given by
+    .. math::
+
+        P_{NL} = \\epsilon_0 \\chi_3 |E|^2 E
+
+    Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for strong nonlinearities. Increasing `numiters` can help with convergence.
+
+    Note
+    ----
+    For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+    is applied separately to the real and imaginary parts, so that the above equation
+    holds when both E and :math:`P_{NL}` are replaced by their real or imaginary parts.
+    The nonlinearity is only applied to the real-valued fields since they are the
+    physical fields.
+
+    Note
+    ----
+    Different field components do not interact nonlinearly. For example,
+    when calculating :math:`P_{NL}_x`, we approximate :math:`|E|^2 \\approx |E_x|^2`.
+    This approximation is valid when the E field is predominantly polarized along one
+    of the x, y, or z axes.
+
+    Example
+    -------
+    >>> medium = Medium(permittivity=2, nonlinear_spec=NonlinearSusceptibility(chi3=1))
+    """
+
+    chi3: float = pd.Field(
+        ..., title="Chi3", description="Chi3 nonlinear susceptibility, in units of um^2 / V^2."
```

Comment:
we should pass the units as a keyword argument to `Field`, eg.

```py
chi3: float = pd.Field(
    ...,
    title=...,
    description=...,
    units="um^2 / V^2",
)
```
We parse this field when setting units in the API reference. Could you add this? Thanks.